Automattic/simplenote-ios
# Connect shared db to widgets

### Fix
This PR is the next stage in adding widgets to Simplenote iOS.  In this PR I have added the things that we need to connect the coreData database (now in the shared directory) to the widgets and intents.

By adding this connection we will be able to query the database for the notes and display them in the widgets, or use them to configure the widgets.

Most of this PR is laying the groundwork for the next steps, so there isn't a lot to see, but I have added some placeholder connections that allow seeing that it is working by trying to configure the note widget.  Note that I have NOT set it up to actually display a note in the widget, only in the configuration.

### Test
1. log into a simplenote account on a device or sim
2. Go to the home screen and long press to edit the home screen layout
3. tap the app button -> Simplenote -> Note Preview widget
4. tap on the newly added widget to get it's configuration settings.
5. you will see a Note option.  Tap on that. 

Notice that this displays the titles of the notes saved in the shared database
Again note that once selected the widget can not currently display the selected note.

### Review
***(Required)*** Add instructions for reviewers.  For example:
> Only one developer is required to review these changes, but anyone can perform the review.

### Release
> These changes do not require release notes.
